src/stock-items/add-stock-item/stock-item-references/stock-item-references.component.tsx

```diff
@@ -0,0 +1,259 @@
+
+  if (isLoading)
+    return (
+      <DataTableSkeleton
+        showHeader={false}
+        rowCount={5}
+        columnCount={5}
+        zebra
+      />
+    );
+
+  return (
+    <FormProvider {...stockReferenceForm}>
+      <DataTable
+        rows={items ?? []}
+        headers={tableHeaders}
+        isSortable={false}
+        useZebraStyles={true}
+        render={({ headers, getHeaderProps, getTableProps }) => (
+          <TableContainer className={styles.referencesTableContainer}>
+            <Table {...getTableProps()} className={styles.referencesTable}>
+              <TableHead>
+                <TableRow>
+                  {headers.map((header) => (
+                    <TableHeader
+                      {...getHeaderProps({
+                        header,
+                        isSortable: false,
+                      })}
+                      style={header.styles}
+                      key={header.key}
+                    >
+                      {header.header?.content ?? header.header}
+                    </TableHeader>
+                  ))}
+                  <TableHeader style={{ width: "70%" }} />
+                </TableRow>
+              </TableHead>
+              <TableBody className={styles.referencesTableBody}>
+                {items?.map((row: StockItemReferenceDTO, index) => (
+                  <StockReferencesRow row={row} key={`${index}-${row?.uuid}`} />
+                ))}
+                <StockReferencesRow row={{}} key="bottom-row" isEditing />
+              </TableBody>
+            </Table>
+          </TableContainer>
+        )}
+      />
+
+      <Button
+        name="save"
+        type="submit"
+        className="submitButton"
+        onClick={handleSaveStockItemReference}
+        kind="primary"
+        renderIcon={Save}
+      >
+        {t("save", "Save")}
+      </Button>
+    </FormProvider>
+  );
+};
+export default StockReferences;
+
+const StockReferencesRow: React.FC<{
+  isEditing?: boolean;
+  row: StockItemReferenceDTO;
+  key?: string;
+}> = ({ isEditing, row, key }) => {
+  const { t } = useTranslation();
+
+  const {
+    control,
+    formState: { errors },
+  } = useFormContext();
+
+  const handleDelete = (e) => {
+    e.preventDefault();
+    deleteStockItemReference(row.uuid).then(
+      () => {
+        showToast({
+          critical: true,
+          title: t("deletePackagingUnitTitle", `Delete StockItem reference`),
+          kind: "success",
+          description: t(
+            "deleteStockItemReferenceMesaage",
+            `StockItem reference deleted Successfully`
+          ),
+        });
+      },
+      (error) => {
+        const err = extractErrorMessagesFromResponse(error);
+        showNotification({
+          title: t(
+            "deleteStockItemReferenceTitle",
+            `Error Deleting a stockitem reference`
+          ),
+          kind: "error",
+          critical: true,
+          description: err.join(","),
+        });
```

Comment:
Can we change `showNotifications and showToast` to use snackbar for uniformity purposes

Reply:
this is noted let me work on this